### PR TITLE
Hyprland enhancements

### PR DESCRIPTION
**Pull request by sewe2000**
I've added what I need for my hyprland rule and fixed some bugs which have prevented me from doing so

### CODE REVIEW

File: pkg/api/sh.go [hunk 1]
```diff
@@ -12,3 +22,33 @@ func ShellCommand(script string) (string, error) {
...
+	err = syscall.Setreuid(uid, uid)
+	if err != nil {
+		return err	
+	}
+
+	const executablesDirectory = "/usr/bin"
+	argv := strings.Split(command, " ")
+	argv[0] = filepath.Join(executablesDirectory, command)
+
+	err = syscall.Exec(argv[0], argv, os.Environ())
```
Comment (cooligus) [inline pkg/api/sh.go:51]: Does it allow using pipes?

Comment (sewe2000) [reply pkg/api/sh.go:51]: No it doesn't. But should we support it in an exec call anyway? Exec is used for replacing the current process

File: pkg/api/sh.go [hunk 2]
```diff
@@ -12,3 +22,33 @@ func ShellCommand(script string) (string, error) {
...
+	if err != nil {
+		return err	
+	}
+
+	err = syscall.Setreuid(uid, uid)
+	if err != nil {
+		return err	
+	}
+
+	const executablesDirectory = "/usr/bin"
```
[the remark this answers is not in the document]
Comment (sewe2000) [reply pkg/api/sh.go:47]: How should the user specify it?

File: pkg/api/sh.go (hunk 1 above)
Comment (sewe2000) [reply pkg/api/sh.go:51]: This function was meant to be used at the end of rules for executing the specified DE or Window Manager

File: pkg/api/sh.go [hunk 3]
```diff
@@ -36,19 +34,13 @@ func Exec(command string) error {
...
 	}
 
-	const executablesDirectory = "/usr/bin"
-	argv := strings.Split(command, " ")
-	argv[0] = filepath.Join(executablesDirectory, command)
-
-	err = syscall.Exec(argv[0], argv, os.Environ())
-	fmt.Println(err.Error())
-	return err
+	return syscall.Exec(command, strings.Split(command, " "), os.Environ())
```
Comment (cooligus) [inline pkg/api/sh.go]: This removes support for strings (`echo "I like cakes"`)

**Review by cooligus — APPROVED** [review]
Check if it applies VRCT
I think it doesn't

**Comment by sewe2000** [comment]
> Check if it applies VRCT

Probably it doesn't

**Pull request merged**